src/compiler/utilities.ts

```diff
@@ -4020,12 +4020,14 @@ namespace ts {
     }
 
     /** Add a value to a set, and return true if it wasn't already present. */
-    export function addToSeen(seen: Map<true>, key: string | number): boolean {
+    export function addToSeen(seen: Map<true>, key: string | number): boolean;
+    export function addToSeen<T>(seen: Map<T>, key: string | number, value: T): boolean;
+    export function addToSeen<T>(seen: Map<T>, key: string | number, value: T = true as any): boolean {

```

Comment:
For my own edification, why not use `{}` as the default value?

Reply:
I've been using `Map<true>` everywhere for sets; to me `Map<{}>` would imply actually being a map from `string` to something, where we didn't specify that something.